tests/unit/test_tool_use_failed_recovery.py

```
@@ -191,3 +195,69 @@ def __init__(self):
+        class Chain:
+            async def ainvoke(self, data):
+                err = Exception("400")
+                err.body = {
+                    "error": {
+                        "message": "Tool choice is none, but model called a tool",
+                        "code": "tool_use_failed",
+                    }
+                }
+                raise err
+
+        with pytest.raises(Exception):
```

Ruff (0.15.10)

[warning] Do not assert blind exception: `Exception`

(B017)